### PR TITLE
Add build-tool dev dependency for fluid-build

### PR DESCRIPTION
**Pull request by curtisman**
fluid-build command needs @fluidframework/build-tools as a dependency. 

Fix [AB#4878](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4878)

### CODE REVIEW

**Review by clarenceli-msft — APPROVED** [review]

File: build-tools/pnpm-lock.yaml [hunk 1]
```diff
@@ -64,7 +67,7 @@ importers:
       '@fluidframework/bundle-size-tools': workspace:*
       '@fluidframework/eslint-config-fluid': ^2.0.0
       '@microsoft/api-extractor': ^7.35.3
-      '@oclif/core': ^2.8.6
+      '@oclif/core': ~2.4.0
```
Comment (tylerbutler) [inline build-tools/pnpm-lock.yaml:70]: Why the oclif downgrade?

File: build-tools/package.json [hunk 2]
```diff
@@ -102,7 +103,8 @@
 		"overrides": {
 			"json5@<1.0.2": "^1.0.2",
 			"json5@>=2.0.0 <2.2.2": "^2.2.2",
-			"qs": "^6.11.0"
+			"qs": "^6.11.0",
+			"@oclif/core": "~2.4.0"
```
Comment (tylerbutler) [inline build-tools/package.json:107]: This shouldn't be needed. If we need to use a different version of oclif then let's update the dependencies of the individual packages.

**Review by tylerbutler — CHANGES_REQUESTED** [review]
I want to understand the oclif downgrade before merge.

File: build-tools/pnpm-lock.yaml (hunk 1 above)
Comment (curtisman) [reply build-tools/pnpm-lock.yaml:70]: because of this:
image

Comment (curtisman) [reply build-tools/pnpm-lock.yaml:70]: The same downgrade happens in the client group as well.

File: build-tools/package.json (hunk 2 above)
Comment (curtisman) [reply build-tools/package.json:107]: Should we down grade build-tool to depend on ~2.4.0 instead of ^2.8.6?

File: build-tools/package.json [hunk 3]
```diff
@@ -70,6 +70,7 @@
 		"@commitlint/config-conventional": "^17.6.5",
 		"@commitlint/cz-commitlint": "^17.5.0",
 		"@fluidframework/build-common": "^2.0.0",
+		"@fluidframework/build-tools": "^0.20.0",
```
Comment (tylerbutler) [inline build-tools/package.json:73]:
```suggestion
		"@fluidframework/build-tools": "workspace:~",
```

Comment (curtisman) [reply build-tools/package.json:73]: that means that `npm run build:fast` doesn't work on first clone?

Comment (curtisman) [reply build-tools/package.json:73]: or maybe we don't want fluid-build to build the build-tools? 😄

**Review by tylerbutler — APPROVED** [review]

File: packages/dds/map/package.json [hunk 4]
```diff
@@ -83,6 +83,7 @@
 		"@fluid-tools/build-cli": "^0.20.0",
 		"@fluidframework/build-common": "^1.2.0",
 		"@fluidframework/build-tools": "^0.20.0",
+		"@fluidframework/build-tools": "^0.20.0",
```
Comment (ChumpChief) [inline packages/dds/map/package.json:86]: FYI, looks like this change duplicated entries in a number of package.jsons @curtisman

**Pull request merged**